File: benchmarks/cases/pascal/pascal.dart
// Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k).
int binom(int n, int k) {
  if (k == 0) {
    return 1;
  } else if (k == n) {
    return 1;
  } else {
    return binom(n - 1, k - 1) + binom(n - 1, k);
  }
}

void main() {
  print(binom(27, 13));
}

File: benchmarks/cases/pascal/pascal.cs
// Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k).
System.Console.WriteLine(Binom(27, 13));

static long Binom(long n, long k)
{
    if (k == 0)
    {
        return 1;
    }
    else if (k == n)
    {
        return 1;
    }
    else
    {
        return Binom(n - 1, k - 1) + Binom(n - 1, k);
    }
}
